### ISSUE TITLE
AMD AMF is not available if the system has both AMD and NVIDIA GPUs.

### ISSUE THREAD

**Issue by Giovani1906** [obsproject/obs-studio#7130]
### Operating System Info

Windows 10

### OBS Studio Version

Other

### OBS Studio Version (Other)

28.0.0-rc1

### OBS Studio Log URL

https://obsproject.com/logs/uivgMSUryGUeHB6e

### Expected Behavior

The AMD AMF encoder to be available alongside NVIDIA NVENC, x264, SVT-AV1 and AOM AV1.

### Current Behavior

Only NVENC, x264, SVT-AV1 and AOM AV1 are available.

### Steps to Reproduce

Go to Settings -> Output -> Streaming/Encoder Settings -> Select an encoder
You can see only NVENC, x264, SVT-AV1 and AOM AV1 as options.

### Anything else we should know?

Disabling the NVIDIA GPU using Windows' Device Manager shows the AMD AMF option again.
`obs-amf-test.exe` recognises the AMD GPU but it doesn't think it supports AVC and HEVC.
```
[0]
is_amd=false
supports_avc=false
supports_hevc=false
[1]
is_amd=true
supports_avc=false
supports_hevc=false
[2]
is_amd=false
supports_avc=false
supports_hevc=false
```
When the NVIDIA GPU is disabled it shows it as being supported.
```
[0]
is_amd=true
supports_avc=true
supports_hevc=true
[1]
is_amd=false
supports_avc=false
supports_hevc=false
```

**Comment by derrod** [comment]
Unfortunately not even AMD's own example app for determining encoding capabilities works realiably. On my AMD notebook (Vega iGPU + RDNA 2 dGPU) it keeps saying the dGPU doesn't support AVC/HEVC unless I run it in Vulkan mode.

**Comment by Giovani1906** [comment]
It seems the issue has been fixed in OBS 29.0.0.
